code/Model/EditableFormField/Validator.php

```diff
@@ -39,31 +39,6 @@ public function php($data)
             return false;
         }
 
-        // Skip unsaved records
```

Comment:
Note to self: It's OK to ditch this logic, because it's CMS side only and it's there to prevent having conditionally displayed, required field.